Subject: Avoid GCC 13's __is_convertible builtin name

Local compatibility patch for libc++ 10.0.1; not an upstream backport.
GCC 13.2 rejects the fallback template's identifier before instantiation.
Rename the internal template only; retain its specializations, completeness
checks and the public std::is_convertible API. The regression forces this
fallback under Clang as well as testing the natural GCC path.

--- a/type_traits
+++ b/type_traits
@@ -1470,29 +1470,29 @@
 template <class _T1, class _T2,
     unsigned _T1_is_array_function_or_void = __is_convertible_imp::__is_array_function_or_void<_T1>::value,
     unsigned _T2_is_array_function_or_void = __is_convertible_imp::__is_array_function_or_void<_T2>::value>
-struct __is_convertible
+struct __oe_is_convertible_fallback
     : public integral_constant<bool,
         __is_convertible_imp::__is_convertible_test<_T1, _T2>::value
     >
 {};
 
-template <class _T1, class _T2> struct __is_convertible<_T1, _T2, 0, 1> : public false_type {};
-template <class _T1, class _T2> struct __is_convertible<_T1, _T2, 1, 1> : public false_type {};
-template <class _T1, class _T2> struct __is_convertible<_T1, _T2, 2, 1> : public false_type {};
-template <class _T1, class _T2> struct __is_convertible<_T1, _T2, 3, 1> : public false_type {};
+template <class _T1, class _T2> struct __oe_is_convertible_fallback<_T1, _T2, 0, 1> : public false_type {};
+template <class _T1, class _T2> struct __oe_is_convertible_fallback<_T1, _T2, 1, 1> : public false_type {};
+template <class _T1, class _T2> struct __oe_is_convertible_fallback<_T1, _T2, 2, 1> : public false_type {};
+template <class _T1, class _T2> struct __oe_is_convertible_fallback<_T1, _T2, 3, 1> : public false_type {};
 
-template <class _T1, class _T2> struct __is_convertible<_T1, _T2, 0, 2> : public false_type {};
-template <class _T1, class _T2> struct __is_convertible<_T1, _T2, 1, 2> : public false_type {};
-template <class _T1, class _T2> struct __is_convertible<_T1, _T2, 2, 2> : public false_type {};
-template <class _T1, class _T2> struct __is_convertible<_T1, _T2, 3, 2> : public false_type {};
+template <class _T1, class _T2> struct __oe_is_convertible_fallback<_T1, _T2, 0, 2> : public false_type {};
+template <class _T1, class _T2> struct __oe_is_convertible_fallback<_T1, _T2, 1, 2> : public false_type {};
+template <class _T1, class _T2> struct __oe_is_convertible_fallback<_T1, _T2, 2, 2> : public false_type {};
+template <class _T1, class _T2> struct __oe_is_convertible_fallback<_T1, _T2, 3, 2> : public false_type {};
 
-template <class _T1, class _T2> struct __is_convertible<_T1, _T2, 0, 3> : public false_type {};
-template <class _T1, class _T2> struct __is_convertible<_T1, _T2, 1, 3> : public false_type {};
-template <class _T1, class _T2> struct __is_convertible<_T1, _T2, 2, 3> : public false_type {};
-template <class _T1, class _T2> struct __is_convertible<_T1, _T2, 3, 3> : public true_type {};
+template <class _T1, class _T2> struct __oe_is_convertible_fallback<_T1, _T2, 0, 3> : public false_type {};
+template <class _T1, class _T2> struct __oe_is_convertible_fallback<_T1, _T2, 1, 3> : public false_type {};
+template <class _T1, class _T2> struct __oe_is_convertible_fallback<_T1, _T2, 2, 3> : public false_type {};
+template <class _T1, class _T2> struct __oe_is_convertible_fallback<_T1, _T2, 3, 3> : public true_type {};
 
 template <class _T1, class _T2> struct _LIBCPP_TEMPLATE_VIS is_convertible
-    : public __is_convertible<_T1, _T2>
+    : public __oe_is_convertible_fallback<_T1, _T2>
 {
     static const size_t __complete_check1 = __is_convertible_check<_T1>::__v;
     static const size_t __complete_check2 = __is_convertible_check<_T2>::__v;
